packages/backend/src/build-disk-image.ts

```diff
@@ -153,64 +155,121 @@ export async function buildDiskImage(build: BootcBuildInfo, history: History, ov
+            const {
+              command: saveCommand,
+              stdout: saveStdout,
+              stderr: saveStderr,
+            } = await extensionApi.process.exec('podman', ['save', '-o', imagePath, build.imageId]);
+            console.log(
+              `Linux build support: Save command: ${saveCommand}\nstdout: ${saveStdout}\nstderr: ${saveStderr}`,
+            );
+            // No 'safe' way to report information at the moment, so we will just increment by 50% as it's done
+            // in two steps anyways. We cannot get a callback of the progress of the exec command yet.
+            progress.report({ increment: 50 });
+
+            // Step 2. Run the command to import and build the image in one command.
+            const command = linuxBuildCommand(buildImageContainer, build, logPath, imagePath);
+            console.log('Linux build support: Running command: ', command);
+            const {
+              command: buildCommand,
+              stdout: buildStdout,
+              stderr: buildStderr,
+            } = await extensionApi.process.exec('sh', ['-c', `${command}`], { isAdmin: true });
+            console.log(
+              `Linux build support: Build command: ${buildCommand}\nstdout: ${buildStdout}\nstderr: ${buildStderr}`,
+            );
           } else {
-            throw new Error('No image to pull');
-          }
-
-          // Step 2. Check if there are any previous builds and remove them
-          progress.report({ increment: 5 });
-          if (buildImageContainer.name) {
-            await containerUtils.removeContainerIfExists(build.engineId, buildImageContainer.name);
-          } else {
-            throw new Error('No container name to remove');
-          }
-
-          // Step 3. Create and start the container for the actual build
-          progress.report({ increment: 6 });
-          build.status = 'running';
-          await history.addOrUpdateBuildInfo(build);
-          const containerId = await containerUtils.createAndStartContainer(build.engineId, buildImageContainer);
+            // Step 1. Pull bootcImageBuilder
+            // Pull the bootcImageBuilder since that
+            // is what is being used to build images within BootC
+            // Do progress report here so it doesn't look like it's stuck
+            // since we are going to pull an image
+            progress.report({ increment: 4 });
+            if (buildImageContainer.Image) {
+              await containerUtils.pullImage(buildImageContainer.Image);
+            } else {
+              throw new Error('No image to pull');
+            }
 
-          // Update the history with the container id that was used to build the image
-          build.buildContainerId = containerId;
-          await history.addOrUpdateBuildInfo(build);
+            // Step 2. Check if there are any previous builds and remove them
+            progress.report({ increment: 5 });
```

Comment:
It feels a bit weird as I had the progress stuck on 50 for most of the time, I feel like an indeterminate would be more appropriate maybe ?

Reply:
yeah :( I will open up another issue for this.

the best path will probably open the image-build.log in a different synchronous process and do continuous reading of it similar to how we stream the container logs.

i'll follow this up in a different PR so this one isn't as big with issue: https://github.com/containers/podman-desktop-extension-bootc/issues/698